walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,185 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.List;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    //If there are no events or no one is attending the event, 
+    // all requests are possible if they are within the range of the whole day.
+
+    // The basic functionality of optional attendees is that if one or more time slots exists 
+    // so that both mandatory and optional attendees can attend, return those time slots. 
+    // Otherwise, return the time slots that fit just the mandatory attendees.
+
+    // Find all events that attendees in the meeting request are already booked for.
+
+    // Mandatory attendees.
```

Comment:
you probably don't need comments explaining these variables, the variable name itself does that